src/main/java/Duke.java

```diff
@@ -1,10 +1,313 @@
+            return;
+        }
+
+        String deadlineBy = parseArgument(commandArgs, COMMAND_DEADLINE_BY_TOKEN);
+        if (isArgumentValueEmpty(deadlineBy)) {
+            printError(ERROR_INVALID_DEADLINE_MESSAGE);
+            return;
+        }
+        recordTask(new Deadline(taskDescription, deadlineBy));
+    }
+
+    /**
+     * Records a new Event task into the global task array.
+     * Ensures the task description and event-at is given in {@code commandArgs}
+     * Fails if any argument value is invalid.
+     *
+     * @param commandArgs this should contain task description and event-at
+     */
+    private static void recordEvent(String commandArgs) {
+        String taskDescription = parseArgument(commandArgs, null);
+        if (isArgumentValueEmpty(taskDescription)) {
+            printError(ERROR_INVALID_TASK_NAME_MESSAGE);
+            return;
+        }
+
+        String eventAt = parseArgument(commandArgs, COMMAND_EVENT_AT_TOKEN);
+        if (isArgumentValueEmpty(eventAt)) {
+            printError(ERROR_INVALID_EVENT_MESSAGE);
+            return;
+        }
+        recordTask(new Event(taskDescription, eventAt));
+    }
+
+    /**
+     * Records the given task into the global task array.
+     * Increments the global task count.
+     *
+     * @param task task object to be recorded
+     */
+    private static void recordTask(Task task) {
+        tasks[numberOfTasks] = task;
+        numberOfTasks++;
+        printStatements(TASK_ADDED_MESSAGE,
+                String.format(DOUBLE_SPACE_PREFIX_STRING_FORMAT, task),
+                String.format(TASK_TOTAL_TASKS_STRING_FORMAT, numberOfTasks));
+    }
+
+    /**
+     * Prints all tasks (tasks are numbered based on addition).
+     */
+    private static void printAllTasks() {
+        printStatement(LIST_TASK_MESSAGE);
+        for (int i = 0; i < numberOfTasks; i++) {
+            Task task = tasks[i];
+            printStatement(String.format("%d.%s", i + 1, task));
+        }
+    }
+
+    /**
+     * Lists all tasks.
+     */
+    private static void listTasks() {
+        if (numberOfTasks == 0) {
+            printStatements(LIST_NO_TASK_MESSAGE);
+        } else {
+            printHorizontalLine();
+            printAllTasks();
+            printHorizontalLine();
+        }
+    }
+
+    /**
+     * Marks a task as done based on the order of the list.
+     *
+     * @param commandArgs this should contain the task number to mark as done
+     */
+    private static void markTaskDone(String commandArgs) {
+        try {
+            String argValue = parseArgument(commandArgs, null);
+            if (argValue == null) {
+                throw new Exception(ERROR_INVALID_TASK_NUMBER_MESSAGE);
+            }
+            int taskNumber = Integer.parseInt(argValue);
+            if (taskNumber > 0 && taskNumber <= numberOfTasks) {
+                int taskIndex = taskNumber - 1;
+                tasks[taskIndex].isDone = true;
```

Comment:
You might want to implement a setDone setter method in the task class to change the isDone variable. The class level variable should be accessed and modified by setter and getter method within the class.

Reply:
Thanks for pointing out, totally forgot to implement the setter.